mcserver/views.py

```diff
@@ -142,258 +135,342 @@ def get_queryset(self):
         if user.is_authenticated and user.id == 1:
             return Session.objects.all().order_by("-created_at")
         return Session.objects.filter(Q(user__id=user.id) | Q(public=True)).order_by("-created_at")
-       
+
     @action(detail=False)
     def api_health_check(self, request):
-        
         return Response({"status": "True"})
-    
+
     @action(
         detail=True,
-        methods=["get","post"],
+        methods=["get", "post"],
     )
     def calibration(self, request, pk):
-        session_path = "/data/{}".format(pk)
-        session = Session.objects.get(pk=pk)
-        trial = session.trial_set.filter(name="calibration").order_by("-created_at")[0]
+        try:
+            if pk == 'undefined':
+                raise ValueError()
+
+            session = get_object_or_404(Session, pk=pk)
+            trial = session.trial_set.filter(name="calibration").order_by("-created_at")[0]
 
-        trial.meta = {
-            "calibration": {
-                cam: val for cam, val in request.data.items()
+            trial.meta = {
+                "calibration": {
+                    cam: val for cam, val in request.data.items()
+                }
             }
-        }
-        trial.save()
+            trial.save()
+        except Http404:
+            raise NotFound('Sorry, we couldn\'t find a session with UUID: ' + pk)
```

Comment:
UUID can become undefined if you refresh steps 2 or 4, i think. should we have a more informative error? Or should this get fixed on the front end that the session id persists through refresh?

Reply:
I think it should be fixed on the frontend, it would be great if the session can persist through a refresh.